tests/testthat/test-rcpp-population-interface.R

```diff
@@ -1,39 +1,140 @@
-library(testthat)
-test_that("Population input settings work as expected", {
-  population <- methods::new(Population)
-  nyears <- 10
-  nages <- 10
-  log_M_length <- nyears * nages
-  population$log_M$resize(log_M_length)
-  purrr::walk(
-    seq_along(1:log_M_length),
-    \(x) population$log_M[x]$value <- -1
-  )
-  population$log_init_naa$resize(nages)
-  purrr::walk(
-    seq_along(1:nages),
-    \(x) population$log_init_naa[x]$value <- 0
-  )
-  population$log_init_naa$set_all_estimable(TRUE)
-  population$nages$set(nages)
-  population$ages$resize(nages)
-  purrr::walk(
-    seq_along(1:nages),
-    \(x) population$ages$set(x - 1, x)
-  )
-  population$nfleets$set(2)
-  population$nseasons$set(1)
-  population$nyears$set(nyears)
-
-  expect_equal(population$get_id(), 1)
-  for (i in 1:(nyears * nages)) {
-    expect_equal(population$log_M[i]$value, -1)
-    expect_false(population$log_M[i]$estimated)
-  }
-  for (i in 1:nyears) {
-    expect_equal(population$log_init_naa[i]$value, 0)
-    expect_true(population$log_init_naa[i]$estimated)
-  }
-
-  clear()
-})
+# Instructions ----
+#' This file follows the format generated by FIMS:::use_testthat_template().
+#' Necessary tests include input and output (IO) correctness [IO
+#' correctness], edge-case handling [Edge handling], and built-in errors and
+#' warnings [Error handling]. See `?FIMS:::use_testthat_template` for more
+#' information. Every test should have a @description tag that takes up just
+#' one line, which will be used in the bookdown report of {testthat} results.
+
+# test_rcpp_population_interface ----
+## IO correctness ----
+test_that("test_rcpp_population_interface() works with correct inputs", {
```

Comment:
Non-blocking thought: Technically, it would be good to test population.evaluate() as well, though I'm not entirely sure what input it requires. I'm leaning toward merging this PR first, and I can look into whether we can add tests for population.evaluate() afterward. Let me know if you have any suggestions.

Reply:
I agree, and it annoyed me too, but population.evaluate() runs the entire population loop which requires everything to be initialized so I assume this can only be tested as part of the integration tests.